webviz_config/_deprecation_store.py

```diff
@@ -0,0 +1,70 @@
+from typing import Any, List, Optional, Dict, Callable, Union
+from dataclasses import dataclass
+
+
+@dataclass(eq=True, frozen=True)
+class DeprecatedPlugin:
+    class_reference: Any
+    short_message: str
+    long_message: str
+
+@dataclass(eq=True, frozen=True)
+class DeprecatedArgument:
+    method_reference: Any
+    method_name: str
+    argument_name: str
+    argument_value: str
+    short_message: str
+    long_message: str
+
+@dataclass(eq=True, frozen=True)
+class DeprecatedArgumentCheck:
+    method_reference: Any
+    method_name: str
+    argument_names: List[str]
+    callback: Callable
+    callback_code: str
+
+
+class DeprecationStore:
+    def __init__(self) -> None:
+        self.stored_plugin_deprecations: Dict[Any, DeprecatedPlugin] = {}
+        self.stored_plugin_argument_deprecations: List[
+            Union[DeprecatedArgument, DeprecatedArgumentCheck]
+        ] = []
+
+    def register_deprecated_plugin(self, deprecated_plugin: DeprecatedPlugin) -> None:
+        """This function is automatically called by the decorator
+        @deprecated_plugin, registering the plugin it decorates.
+        """
+        self.stored_plugin_deprecations[
+            deprecated_plugin.class_reference
+        ] = deprecated_plugin
+
+    def register_deprecated_plugin_argument(
+        self, deprecated_plugin_argument: Union[DeprecatedArgument, DeprecatedArgumentCheck]
+    ) -> None:
+        """This function is automatically called by the decorator
+        @deprecated_plugin_arguments, registering the __init__ function it decorates.
+        """
+        self.stored_plugin_argument_deprecations.append(deprecated_plugin_argument)
+
+    def get_stored_plugin_deprecation(self, plugin: Any) -> Optional[DeprecatedPlugin]:
+        if plugin in self.stored_plugin_deprecations:
+            return self.stored_plugin_deprecations[plugin]
+        return None
+
+    def get_stored_plugin_argument_deprecations(
+        self, method: Callable
+    ) -> List[Union[DeprecatedArgument, DeprecatedArgumentCheck]]:
+        deprecations: List[Union[DeprecatedArgument, DeprecatedArgumentCheck]] = []
+        for stored in self.stored_plugin_argument_deprecations:
+            if stored.method_reference == method:
+                deprecations.append(stored)
+
+        return deprecations
```

Comment:
```suggestion
        return [stored for stored in self.stored_plugin_argument_deprecations if stored.method_reference == method]
```
Not tested, but maybe a shorter version like this is possible.